hyperspy/axes.py

```diff
@@ -655,6 +675,18 @@ def _append_axis(self, *args, **kwargs):
         axis.axes_manager = self
         self._axes.append(axis)
 
+    def _on_navigate(self):
+        self._update_attributes()
+        self._events.navigated.trigger()
+
+    def _on_nav_axes_changed(self):
+        self._update_attributes()
+        self._events.axes_changed.trigger()
+
+    def _on_axes_size_changed(self):
+        self._update_attributes()
+        self._events.axes_changed.trigger()
```

Comment:
`axes_size_changed`? Otherwise, if you don't want to be that specific, just connecting to the `axis_changed` events should do.